src/main/java/com/android/volley/cronet/CronetHttpStack.java

```diff
@@ -0,0 +1,248 @@
+                                new HttpResponse(
+                                        urlResponseInfo.getHttpStatusCode(),
+                                        headers,
+                                        bytesReceived.toByteArray());
+                        callback.onSuccess(response);
+                    }
+
+                    @Override
+                    public void onFailed(
+                            UrlRequest urlRequest,
+                            UrlResponseInfo urlResponseInfo,
+                            CronetException e) {
+                        if (urlResponseInfo != null) {
+                            List<Header> headers =
+                                    getHeaders(urlResponseInfo.getAllHeadersAsList());
+                            HttpResponse response =
+                                    new HttpResponse(
+                                            urlResponseInfo.getHttpStatusCode(),
+                                            headers,
+                                            bytesReceived.toByteArray());
+                            callback.onSuccess(response);
+                        } else {
+                            callback.onError(e);
+                        }
+                    }
+                };
+        UrlRequest.Builder builder =
+                mCronetEngine
+                        .newUrlRequestBuilder(request.getUrl(), urlCallback, mCallbackExecutor)
+                        .allowDirectExecutor();
+        try {
+            setHttpMethod(request, builder);
+            setRequestHeaders(request, additionalHeaders, builder);
+            UrlRequest urlRequest = builder.build();
+            urlRequest.start();
+        } catch (AuthFailureError authFailureError) {
+            callback.onAuthError(authFailureError);
+        }
+    }
+
+    @VisibleForTesting
+    public static List<Header> getHeaders(List<Map.Entry<String, String>> headersList) {
+        List<Header> headers = new ArrayList<>();
+        for (Map.Entry<String, String> header : headersList) {
+            headers.add(new Header(header.getKey(), header.getValue()));
+        }
+        return headers;
+    }
+
+    /** Sets the connection parameters for the UrlRequest */
+    private void setHttpMethod(Request<?> request, UrlRequest.Builder builder)
+            throws AuthFailureError {
+        switch (request.getMethod()) {
+            case Request.Method.DEPRECATED_GET_OR_POST:
+                // This is the deprecated way that needs to be handled for backwards compatibility.
+                // If the request's post body is null, then the assumption is that the request is
+                // GET.  Otherwise, it is assumed that the request is a POST.
+                byte[] postBody = request.getPostBody();
+                if (postBody != null) {
+                    builder.setHttpMethod("POST");
+                    addBodyIfExists(postBody, builder);
+                }
```

Comment:
should this have an else with builder.setHttpMethod("GET")? This is what it sounds like should happen from your comment above. Or do you not have to because it's the default? Either way, I think you should be explicit – either set it to "GET" (like below) or add to the comment saying that's the default

Reply:
It is the default, but I'll be explicit and set it.